src/main/frontend/modules/outliner/core.cljs

```diff
@@ -25,8 +25,14 @@
 
 (defn block
   [m]
-  (assert (map? m) (util/format "block data must be map, got: %s %s" (type m) m))
-  (->Block m))
+  (assert (or (map? m) (de/entity? m)) (util/format "block data must be map or entity, got: %s %s" (type m) m))
+  (if (de/entity? m)
```

Comment:
I didn't see a new use of this fn. Curious what this does

Reply:
If you don't have a repro for this, I'd recommend removing it. It's more complexity that you and no one else will  understand

Reply:
I'll give it more tests tomorrow to find the cause, I'm sure there'll be some bugs if we just remove it.

Reply:
You can reproduce the bug by changing it to `(assert (map? m))` and unindent in any child block.
The reason is that `get-block-immediate-children` returns entities instead of maps to improve performance.

https://github.com/logseq/logseq/blob/master/src/main/frontend/modules/outliner/core.cljs#L231-L232
https://www.loom.com/share/08bacdfa706a4de599198d3b5ecf9d64

Reply:
That makes sense. Glad to hear it's used